specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2022-08-31-preview/cloudhsm.json

```diff
@@ -0,0 +1,1017 @@
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/CHsmError"
+        }
+      },
+      "description": "The Cloud HSM Cluster error details",
+      "type": "object"
+    },
+    "CHsmError": {
+      "properties": {
+        "code": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The error code."
+        },
+        "message": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The error message."
+        }
+      },
+      "type": "object",
+      "description": "Error details"
+    },
+    "CloudHsmClusterListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CloudHsmCluster"
+          },
+          "description": "The list of Cloud HSM Clusters."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of Cloud HSM Clusters."
+        }
+      },
+      "description": "List of Cloud HSM Clusters",
+      "type": "object"
+    },
+    "PrivateEndpointConnection": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/PrivateEndpointConnectionProperties",
+          "x-ms-client-flatten": true,
+          "description": "Resource properties."
+        },
+        "etag": {
+          "type": "string",
+          "description": "Modified whenever there is a change in the state of private endpoint connection."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "description": "The private endpoint connection resource.",
+      "x-ms-azure-resource": true
+    },
+    "PrivateEndpointConnectionProperties": {
+      "type": "object",
+      "properties": {
+        "privateEndpoint": {
+          "$ref": "#/definitions/PrivateEndpoint",
+          "description": "The private endpoint resource."
+        },
+        "privateLinkServiceConnectionState": {
+          "$ref": "#/definitions/PrivateLinkServiceConnectionState",
+          "description": "A collection of information about the state of the connection between service consumer and provider."
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/PrivateEndpointConnectionProvisioningState",
+          "description": "The provisioning state of the private endpoint connection resource."
+        },
+        "groupIds": {
+          "description": "The group ids for the private endpoint resource.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "readOnly": true
+        }
+      },
+      "required": [
+        "privateLinkServiceConnectionState"
+      ],
+      "description": "Properties of the private endpoint connection."
+    },
+    "PrivateEndpoint": {
```

Comment:
>PrivateEndpoint [](http://example.com/codeflow?start=5&length=15)

Why defining PrivateEndpoint and PrivateLink related definitions here locally? These are available here: [https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/privatelinks.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/privatelinks.json).

Reply:
update swagger to reuse most of Private endpoint definitions except from these:
PrivateEndpointConnection  ---> Our object has "etag" property
PrivateEndpointConnectionProvisioningState  ---> We have additional states 
PrivateEndpointConnectionListResult --->   Our object has "nextLink" property